### PR TITLE
[WIP] Named endpoints

### PR DESCRIPTION
**Pull request by LinusU**
Per discussion in apiaryio/api-blueprint#196.

For some reason it keeps parsing the section as copy instead of resource. No idea why thought, would appreciate some fresh :eyes: on it.

### CODE REVIEW

**Comment by zdne** [comment]
@LinusU Hey, thanks for the submission! 

I think you want to check the `sectionType()` method here https://github.com/LinusU/snowcrash/blob/named-endpoints/src/ResourceParser.h#L196 – it is basically used to figure out whether a markdown node is recognized as a resource section (= you want to match the REGEX there)

**Comment by LinusU** [comment]
Ahhh, one line! :laughing:

```diff
@@ -207,6 +204,7 @@ namespace snowcrash {
                 TrimString(subject);
 
                 if (RegexMatch(subject, NamedResourceHeaderRegex) ||
+                    RegexMatch(subject, NamedEndpointHeaderRegex) ||
                     RegexMatch(subject, ResourceHeaderRegex)) {
                     return ResourceSectionType;
                 }
```

Thanks for the help! :beers:

I'm pushing a rebased version now

**Comment by zdne** [comment]
@LinusU also can you add tests for edge cases like 

```apib
# Endpoint 1  [GET /e1]

**Comment by LinusU** [comment]
Sure, I'll add them now!

**Comment by LinusU** [comment]
Done and done, was that how you meant?

**Comment by LinusU** [comment]
Ahh, I read your comment in the other thread now. Do you want it to merge it all into one resource with three actions?

File: snowcrash.gyp [hunk 1]
```diff
@@ -156,6 +156,7 @@
         'test/test-MSONTypeSectionParser.cc',
         'test/test-MSONUtility.cc',
         'test/test-MSONValueMemberParser.cc',
+        'test/test-NamedEndpoint.cc',
```
Comment (pksunkara) [inline snowcrash.gyp]: I don't think these test cases should be in a separate file. Please add them in `test-ActionParser.cc`.

File: src/ResourceParser.h [hunk 2]
```diff
@@ -336,6 +334,29 @@ namespace snowcrash {
...
         }
 
+        /**
+         * \brief Parse the current node as an action
+         */
+        static MarkdownNodeIterator processNestedAction(const MarkdownNodeIterator& node,
+                                  const MarkdownNodes& siblings,
+                                  SectionParserData& pd,
+                                  SectionLayout& layout,
+                                  const ParseResultRef<Resource>& out) {
```
Comment (pksunkara) [inline src/ResourceParser.h]: The indentation of the above 4 lines should align with the first argument in the function.

File: src/ResourceParser.h [hunk 3]
```diff
@@ -49,21 +52,15 @@ namespace snowcrash {
 
                 // Make this section an action
                 if (!captureGroups[2].empty()) {
+                    return processNestedAction(node, siblings, pd, layout, out);
```
Comment (pksunkara) [inline src/ResourceParser.h]: I don't think the `siblings` given here is correct. If you look at the old code, they are `node->parent().children()`

Comment (LinusU) [inline src/ResourceParser.h]: I thought that it would yield the same result, and since all the tests passed I figured I was right... Want me to revert?

Comment (pksunkara) [inline src/ResourceParser.h]: Yes, please keep it `node->parent().children()`.

**Comment by LinusU** [comment]
Should be good to go :+1:

**Comment by LinusU** [comment]
Thanks for the merge :tada: :v:
